packages/renderer/server/renderVideo.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  concatenateMedia,
+  createSilentAudioFile,
+  doesFileExist,
+  getVideoDuration,
+  mergeAudioWithVideo,
+} from '@revideo/ffmpeg';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import puppeteer, {Browser, BrowserLaunchArgumentOptions} from 'puppeteer';
+import * as readline from 'readline';
+import {v4 as uuidv4} from 'uuid';
+import {ViteDevServer, createServer} from 'vite';
+import {rendererPlugin} from './RendererPlugin';
+
+function buildUrl(
+  port: number,
+  fileName: string,
+  workerId: number,
+  totalNumOfWorkers: number,
+  range: [number, number] = [0, Infinity],
+  hiddenFolderId: string,
+) {
+  return `http://localhost:${port}/render?fileName=${fileName}&workerId=${workerId}&totalNumOfWorkers=${totalNumOfWorkers}&startInSeconds=${range[0]}&endInSeconds=${range[1]}&hiddenFolderId=${hiddenFolderId}`;
+}
+
+interface RenderVideoSettings {
+  // Name of the video file
+  name?: string;
+
+  // Start and end in seconds
+  range?: [number, number];
+
+  puppeteer?: BrowserLaunchArgumentOptions;
+  numWorkers?: number;
+}
+
+export const renderVideo = async (
+  configFile: string,
+  params?: Record<string, unknown>,
+  settings: RenderVideoSettings = {},
+) => {
+  console.log('Rendering...');
+
+  const resolvedConfigPath = path.resolve(process.cwd(), configFile);
+  const projectName = settings.name ?? 'project';
+  const hiddenFolderId = uuidv4();
+
+  /**
+   * TODO: Change this in the future after more testing.
+   * We still need to concate the videos together after rendering.
+   */
+
+  const numOfWorkers = settings.numWorkers ?? 1;
+  const launchPromises = [];
+  for (let i = 0; i < numOfWorkers; i++) {
+    launchPromises.push(
+      initBrowserAndServer(
+        9000 + i,
+        resolvedConfigPath,
+        params,
+        settings.puppeteer,
+      ),
+    );
+  }
+
+  const launchedServices = await Promise.all(launchPromises);
+  const renderPromises = [];
+  for (let i = 0; i < numOfWorkers; i++) {
+    const {browser, server, port} = launchedServices[i];
+    const url = buildUrl(
+      port,
+      `${projectName}-${i}`,
+      i,
+      numOfWorkers,
+      settings.range,
+      hiddenFolderId,
+    );
+
+    renderPromises.push(renderVideoOnPage(browser, server, url));
+  }
```

Comment:
I think we're losing time here. Couldn't a browser start rendering right after it's ready? Right now the first one needs to wait for all the others before it can start.